### DISCUSSION TITLE
Automating Downloads on Nicotine+ Based on Wishlist ?

### DISCUSSION THREAD

**Discussion by Kalcinator** [nicotine-plus/nicotine-plus#2618] [Ideas]
Hey everyone!

I've been exploring free music on Nicotine+ for a while now and am aiming to (try to) automate my downloads.

I'm using Python, though I'm pretty new to it, I'm super curious about your work and Nicotine+ is actually the first "real" software in Python I looked through.

Here's my approach: 
I'll pull tracks from my Nicotine+ wishlist and download everything new with the highest quality versions available. 
Before any download, I'll cross-reference with my local collection to skip songs I already have. 
If a better quality is found, I'll replace the lower ones.
Eventually I want to log every download or change.

Got advice or tips ?

Thanks for your hard free work !!

**Comment by slook** [comment]
Please give a thumbs up in issue https://github.com/nicotine-plus/nicotine-plus/issues/1955

> Got advice or tips ?

Before you start any development on this, wait until https://github.com/nicotine-plus/nicotine-plus/issues/2522 and https://github.com/nicotine-plus/nicotine-plus/discussions/1604 have been implemented first.
